js/src/data/index.js

```diff
@@ -21,6 +25,17 @@ registerStore( STORE_KEY, {
 	reducer,
 } );
 
+apiFetch.use(
+	createErrorResponseCatcher( ( response ) => {
+		if ( response.status === 401 ) {
+			getHistory().replace( getReconnectAccountsUrl() );
```

Comment:
:scroll:
I have a doubt about the UX here. Do we really want to replace the history = disallow a user to go back to retry after they re-connect the account?

Personally, as a heavy back button user, I'd prefer `pushState` here

Reply:
Currently, when going back to the page which had got error responses, we don't have a wide implementation to retry loading data. Thus, there's no API request to trigger 401 error response and re-redirect user to reconnection page again. It would display a page without any error message (#849) and no available data to render. Compare to that, I think it should be relatively acceptable to use `replace` for now?

Reply:
I didn't mean retry loading data. The flow I was referring to was:
1. User: "I'd like to go to the dashboard to see my stats"
2. User: visits `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`
2. GLA: Fetch the stats, 401, redirects to reconnect page
4. ⋮
5. User: reconnects Google  account
6. GLA: "Account successfully reconnected"
7. User: "Cool, I'd like to get back to where I was at" Clicks "back button" n times/right-click to pick from history.

> Compare to that, I think it should be relatively acceptable to use `replace` for now?

Definitely, probably the flow could change in the meantime anyway. And `replace` is better than the empty page.

Reply:
> 1\. User: "I'd like to go to the dashboard to see my stats"
> 6\. GLA: "Account successfully reconnected"

After merging #838, users will be redirected to the dashboard page at step 6. But no matter the original page users wanted to view at step 1, they're always redirected to the dashboard page after reconnected.

I was thinking about adding a UX improvement by:

1. When the 401 is detected, store the current page path in `sessionStorage` before redirecting to reconnect
2. After Google account successfully reconnected, bring users back to the page stored in `sessionStorage` if it exists. Otherwise, bring them to the default (dashboard) page.

By this flow, it'd still using `replace` but the concern you mentioned could be handled in a more user-friendly way. WDYT @tomalec ?